plugins/shortcuts/mpris.js

```diff
@@ -34,6 +34,35 @@ function registerMPRIS(win) {
 		let currentSeconds = 0;
 		ipcMain.on('timeChanged', (_, t) => currentSeconds = t);
 
+		let currentLoopStatus = undefined;
+		let manuallySwitchingStatus = false;
+		ipcMain.on("repeatChanged", (_, mode) => {
+			if (manuallySwitchingStatus)
+				return;
+
+			if (mode == "Repeat off")
+				currentLoopStatus = "None";
+			else if (mode == "Repeat one")
+				currentLoopStatus = "Track";
+			else if (mode == "Repeat all")
+				currentLoopStatus = "Playlist";
+
+			player.loopStatus = currentLoopStatus;
+		});
+		player.on("loopStatus", (status) => {
+			// switchRepeat cycles between states in that order
+			const switches = ["None", "Playlist", "Track"];
+			const curIdx = switches.indexOf(currentLoopStatus);
+			const newIdx = switches.indexOf(status);
+
+			// Get a delta in the range [0,2]
+			const delta = ((newIdx - curIdx) % 3 + 3) % 3;
+
+			manuallySwitchingStatus = true;
+			songControls.switchRepeat(delta);
```

Comment:
This approach is a bit of a hack (it presses the button the appropriate number of times to go to the desired state), but I can't see a better way.

Reply:
I've tried making it work using the youtube api, but its weird and doesn't visually change the button state

So your solution is probably the best bet